### DISCUSSION TITLE
Addition of XMSS/LMS

### DISCUSSION THREAD

**Discussion by dalGob** [open-quantum-safe/liboqs#1875] [General]
I was wondering about the reasons behind the addition of stateful signatures like XMSS/LMS to liboqs. I realize they have been mentioned in CNSA 2.0, but considering the security issues behind the stateful nature, I was wondering if there were other reasons.

**Comment by ashman-p** [comment]
> I was wondering about the reasons behind the addition of stateful signatures like XMSS/LMS to liboqs. I realize they have been mentioned in CNSA 2.0, but considering the security issues behind the stateful nature, I was wondering if there were other reasons.
Hi, the consideration was/is that it aligns with OQS's mission of providing space for furthering the development, and experimentation of PQ safe crypto. The security considerations had also been discussed and addressed by disabling key and signature generation. By default the library just does signature verification.
HTH.

**Comment by dalGob** [comment]
Thanks for the response!
